src/mlpack/tests/dbscan_test.cpp

```diff
@@ -201,6 +213,36 @@ TEST_CASE("OutlierSingleModeTest", "[DBSCANTest]")
   REQUIRE(assignments[101] == SIZE_MAX);
 }
 
+/**
+ * Check that outliers are properly labeled as noise
+ * in the case of float32.
+ */
+TEST_CASE("Float32OutlierSingleModeTest", "[DBSCANTest]")
+{
+  arma::Mat<float> points(2, 200, arma::fill::randu);
+
+  // Add 3 outliers.
+  points.col(15) = arma::Col<float>("10.3 1.6");
+  points.col(45) = arma::Col<float>("-100 0.0");
+  points.col(101) = arma::Col<float>("1.5 1.5");
+
+  DBSCAN<RangeSearch<
+         EuclideanDistance,
+         arma::Mat<float>,
+         FloatKDTree>,
+         OrderedPointSelection,
+         arma::Mat<float>> d(0.1, 3, false);
```

Comment:
One ugliness here is that the `MatType` parameter doesn't automatically get forwarded to `RangeSearch`.  One way around this would be to expose the parameter inside `RangeSearchType`, and take `DBSCAN`s `MatType` parameter to have a default value of `RangeSearchType::MatType`.  That way, a user only needs to specify the `MatType` as part of `RangeSearchType`, instead of in two places.  Maybe it is not the best solution, but what do you think? I would like to avoid requiring users to specify `float` or `fmat` twice.

Reply:
Yes, well, I thought about this, but the main reason for me is that most of the functions inside DBSCAN required `MatType` but not the class itself, so I ended up refactoring DBSCAN instead and adding a `MatType` to the class

one solution could be as you proposed, in this case, MatType would not be necessary in DBSCAN.

Would that make sense to you?